examples/examples/jsonrpsee_as_service.rs

```diff
@@ -58,30 +57,33 @@ struct Metrics {
 	http_connections: Arc<AtomicUsize>,
 }
 
+#[derive(Clone)]
 struct AuthorizationMiddleware<S> {
 	headers: HeaderMap,
 	inner: S,
 	#[allow(unused)]
 	transport_label: &'static str,
 }
 
-#[async_trait]
 impl<'a, S> RpcServiceT<'a> for AuthorizationMiddleware<S>
 where
-	S: Send + Sync + RpcServiceT<'a>,
+	S: Send + Clone + Sync + RpcServiceT<'a>,
```

Comment:
It didn't occur to me before but I guess, since you only need to take `&self`, that `S` could be wrapped in an `Arc` or something if you want to be able to cheaply clone it while avoiding the `Clone` bounds on `S`.

But `Clone` bounds on `S` might be better because it doesn't dictate how `S` is made cloneable and so they can opt to wrap it in  `Arc` or maybe just copy it or whatever if it's cheap etc.

Reply:
good idea, it's probably a good practice to wrap in an Arc instead cloning the entire service

Reply:
But I mean that as it is right now, it's up to the user whether they want to wrap the inner service in an Arc or whether they don't need to, so I do sortof like how it is now :)

Reply:
yepp, I was referring to the examples that we should point to folks to wrap it an Arc instead of cloning the entire service because it can be expensive such if one put HTTP headers or something heavier in there